src/video/wayland/SDL_waylandvideo.c

```diff
@@ -617,10 +620,15 @@ static void display_handle_done(void *data,
         if (video->viewporter != NULL) {
             SDL_AddDisplayMode(dpy, &native_mode);
         } else {
-            /* ...if not, expose the integer scale desktop modes down to 1.0. */
+            /* ...if not, expose some smaller, integer scaled resolutions. */
             int i;
-            for (i = (int)driverdata->scale_factor - 1; i > 0; --i) {
+            const int base_pixel_w = desktop_mode.pixel_w / (int)desktop_mode.display_scale;
+            const int base_pixel_h = desktop_mode.pixel_h / (int)desktop_mode.display_scale;
+            for (i = 1; i < (int)desktop_mode.display_scale; ++i) {
```

Comment:
Is it intentional that you could do i = { 1, 2, 3 } when display_scale = 2.6, based on floating point rounding mode?
Do you want explicit rounding here, using SDL_floorf() or SDL_ceilf()?

Reply:
If we get here, we can be 100% certain that the scale value is an integer.  Wayland only exposes integer scale values for displays, so we have to calculate the fractional value ourselves, which isn't done if the viewport protocol isn't present, and the absence of the viewport protocol is the only way we end up in this particular section of code.

Reply:
Gotcha. Okay, merging!